CONTRIBUTING.md

```diff
@@ -652,27 +652,31 @@ Refer datalad/config.py for information on how to add these environment variable
 - `update-changelog`: uses above `linkissues-changelog` and updates .rst changelog
 - `release-pypi`: ensures no `dist/` exists yet, creates a wheel and a source distribution and uploads to pypi.
 
-## Changelog section
-
-For the upcoming release use this template
-
-```markdown
-## 0.14.6 (??? ??, 2021) -- will be better than ever
-
-bet we will fix some bugs and make a world even a better place.
-
-### Major refactoring and deprecations
-
-- hopefully none
-
-### Fixes
-
-?
-
-### Enhancements and new features
-
-?
-```
+## Releasing with GitHub Actions, auto, and pull requests
+
+New releases of datalad are created via a GitHub Actions workflow built
+around [`auto`](https://github.com/intuit/auto).  Whenever a pull request is
+merged into `maint` that has the "`release`" label, `auto` updates the
+changelog based on the pull requests since the last release, commits the
+results, tags the new commit with the next version number, and creates a GitHub
+release for the tag.  This in turn triggers a job for building an sdist & wheel
+for the project and uploading them to PyPI.
+
+### Labelling pull requests
+
+The section that `auto` adds to the changelog on a new release consists of the
+titles of all pull requests merged into master since the previous release,
+organized by label.  `auto` recognizes the following PR labels:
+
+- `minor` — for changes corresponding to an increase in the minor version
+  component
+- `patch` — for changes corresponding to an increase in the patch/micro version
+  component; this is the default label for unlabelled PRs
+- `internal` — for changes only affecting the internal API
+- `documentation` — for changes only affecting the documentation
+- `tests` — for changes to tests
+- `dependencies` — for updates to dependency versions
+- `performance` — for performance improvements
```

Comment:
@jwodder I think we are missing also a section on how to label issues so that `auto` could correctly catalog into `Enhancement`, `Bug Fix` etc.  Does it do it based on issues closed by a PR or PR should have one of those labels?

Reply:
`auto` only pays attention to PRs, not issues.

Reply:
or it does it on minor (enhancement)/patch (bug fix) distinction?

Reply:
What do you mean?

Reply:
How does it decide either changelog entry goes to bugfix or enhancement subsection?

Reply:
Based on the labels.  `minor` is "Enhancement", `patch` is "Bugfix".  You can see a dump of the label-to-section mappings by running `auto info -v` in a clone of the repository.

Reply:
then we should allow for `patch` to also be in PRs against `master` - there might be bugfixes for master-only relevant content, although then the question is -- should they be in the changelog at all.  Is there a way to annotate a PR to not participate in the changelog?

Reply:
I don't think so.

Reply:
ok, I have tried locally by merging this branch into `maint` and trying out `auto` commands:
- `info -v` was not happy about labels (because of my removal of `major` I guessed) so I reran `create-labels` and it made it happier
- `auto version` outputs nothing, although running in `-v` says:
```
✔  success   Calculated SEMVER bump: 

```

any idea why no new version is produced? may be because merge was local? `git describe` says 0.14.4-59-g484e60b80
- FTR: my local `auto` is v10.25.0
- running ` changelog` produces changelog but it seems to pick up individual commits, and for PRs -- merge commits not PR titles (i.e. e.g. `Merge pull request #5671 from kyleam/update-fetch-fail`),

<details>
<summary>here is what is produced</summary> 

# (Thu Jun 03 2021)

#### 🐛 Bug Fix

- Give auto commits a descriptive name and defunct ATM email ([@yarikoptic](https://github.com/yarikoptic))
- BF: try to mitigate not listing of importlib-metadata in install requires for now ([@yarikoptic](https://github.com/yarikoptic))
- RF/BF: determine version in _datalad_build_support via importlib_metadata ([@yarikoptic](https://github.com/yarikoptic))
- BF(TST): adjust test_version to match expectations from switching to auto ([@yarikoptic](https://github.com/yarikoptic))
- BF: account for annex now "scanning for annexed" instead of "unlocked" files [#5705](https://github.com/datalad/datalad/pull/5705) ([@yarikoptic](https://github.com/yarikoptic))
- Update .autorc ([@jwodder](https://github.com/jwodder))
- Pass token to gh ([@jwodder](https://github.com/jwodder))
- Fix add-label workflow syntax ([@jwodder](https://github.com/jwodder))
- remove documentation of `major` label which we will not have for now ([@yarikoptic](https://github.com/yarikoptic))
- Update CONTRIBUTING.md ([@jwodder](https://github.com/jwodder))
- Make CHANGELOG formatting auto-friendly ([@jwodder](https://github.com/jwodder))
- Add workflow for adding labels to PRs ([@jwodder](https://github.com/jwodder))
- Release only from maint ([@yarikoptic](https://github.com/yarikoptic))
- interface: Don't repeat custom summary for non-generator results [#5688](https://github.com/datalad/datalad/pull/5688) ([@kyleam](https://github.com/kyleam))
- RF: just pip install datalad-installer [#5676](https://github.com/datalad/datalad/pull/5676) ([@yarikoptic](https://github.com/yarikoptic))
- DOC: addurls.extract: Drop mention of removed 'stream' parameter ([@kyleam](https://github.com/kyleam))
- Merge pull request #5674 from kyleam/test-addurls-copy-fix [#5674](https://github.com/datalad/datalad/pull/5674) ([@kyleam](https://github.com/kyleam))
- RF/BF: do not make clone shallow in appveyor ([@yarikoptic](https://github.com/yarikoptic))
- upgrade with devel version of versioneer to get sensible version from shallow clones ([@yarikoptic](https://github.com/yarikoptic))
- RF: return back datalad.version to not break "API" for maint ([@yarikoptic](https://github.com/yarikoptic))
- Merge pull request #5663 from kyleam/status-ds-equal-path [#5663](https://github.com/datalad/datalad/pull/5663) ([@kyleam](https://github.com/kyleam))
- Merge pull request #5671 from kyleam/update-fetch-fail [#5671](https://github.com/datalad/datalad/pull/5671) ([@kyleam](https://github.com/kyleam))
- Set up auto ([@jwodder](https://github.com/jwodder))
- Switch to versioneer ([@jwodder](https://github.com/jwodder))
- Merge pull request #5664 from kyleam/addurls-better-url-parts-error [#5664](https://github.com/datalad/datalad/pull/5664) ([@kyleam](https://github.com/kyleam))
- Merge pull request #5661 from kyleam/sphinx-fix-plugin-refs [#5661](https://github.com/datalad/datalad/pull/5661) ([@kyleam](https://github.com/kyleam))
- download-url: Set up datalad special remote if needed [#5648](https://github.com/datalad/datalad/pull/5648) ([@kyleam](https://github.com/kyleam) [@yarikoptic](https://github.com/yarikoptic))

#### ⚠️ Pushed to `maint`

- Merge branch 'gh-4999' into maint ([@yarikoptic](https://github.com/yarikoptic))
- MNT: Post-release dance ([@kyleam](https://github.com/kyleam))

#### 🏠 Internal

- MNT: setup.py: Temporarily avoid Sphinx 4 [#5649](https://github.com/datalad/datalad/pull/5649) ([@kyleam](https://github.com/kyleam))

#### 🧪 Tests

- Revert "TST: test_partial_unlocked: Document and avoid recent git-annex failure" [#5651](https://github.com/datalad/datalad/pull/5651) ([@kyleam](https://github.com/kyleam))

#### Authors: 3

- John T. Wodder II ([@jwodder](https://github.com/jwodder))
- Kyle Meyer ([@kyleam](https://github.com/kyleam))
- Yaroslav Halchenko ([@yarikoptic](https://github.com/yarikoptic))

---

</details>

is there a way to make it pickup PR titles and not bother about individual commits?  may be individual commits are only the ones for this PR though which we did not have yet merged, so at least it could provide explanation for those.  but PRs -- we should do smth about that

Reply:
The `.autorc` file is configured to only make a release when a PR with the `release` label has been merged since the last release; if there is no such PR, `auto version` will not output anything.  You can try setting `onlyPublishWithReleaseLabel` in `.autorc` to false and then running `auto version`, and that should output a semver level.

I don't know why the changelog is reporting individual commits.  It may be in part because you merged this branch locally rather than merging the PR, but that does not appear to explain all of the commits.

Reply:
> `onlyPublishWithReleaseLabel` in `.autorc` to false and then running `auto version`, and that should output a semver level.

confirming that.

ok -- let's forget about commits for now.  Why we do not get descriptive summary descriptions for the merged PRs?  in dandi-cli we get those AFAIK.  May be yet another side-effect of not having a proper "release PR merge", but it would be nice to know that for sure so we do not end up with a very deficient changelog  we would need to fix up.

Reply:
I don't know.  What branch are you on when running `auto`?  When I try to run `auto` from `maint` after copying over `.autorc`, I just get a bunch of API rate limit errors.

Reply:
I also merge this PR branch into maint (locally, add if we merged the PR) to run - otherwise I guess auto might be taking too many PRs to consider etc.